api_app/script_analyzers/file_analyzers/xlm_macro_deobfuscator.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from api_app.script_analyzers.classes import FileAnalyzer
+from celery.exceptions import SoftTimeLimitExceeded
+import subprocess
+from pathlib import Path
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class XlmMacroDeobfuscator(FileAnalyzer):
+    def set_config(self, additional_config_params):
+        self.passwords_to_check = []
+        additional_passwords_to_check = additional_config_params.get(
+            "passwords_to_check", []
+        )
+        if isinstance(additional_passwords_to_check, list):
+            self.passwords_to_check.extend(additional_passwords_to_check)
+        elif isinstance(additional_passwords_to_check, str):
+            self.passwords_to_check.append(additional_passwords_to_check)
+
+    def run(self):
+        results = {}
+        try:
+            if not self.passwords_to_check:
+                results = self.decrypt()
+            else:
+                for password in self.passwords_to_check:
+                    results = self.decrypt(password)
+                    if results:
+                        break
+            if not results:
+                results["errors"] = "Unable to deobfuscate"
+        except SoftTimeLimitExceeded as e:
+            error_message = (
+                f"job_id:{self.job_id} analyzer:{self.analyzer_name} md5:{self.md5}"
+                f"filename: {self.filename}. Soft Time Limit Exceeded Error {e}"
+            )
+            logger.error(error_message)
+            self.report["errors"].append(str(e))
+            self.report["success"] = False
+        return results
+
+    def decrypt(self, xlmpassword=""):
+        cmd = [
+            "xlmdeobfuscator",
+            "--file",
+            self.filepath,
+            "--export-json",
+            "result.json",
+        ]
+        pwd_cmd = ["--password", xlmpassword] if xlmpassword else []
+        subprocess.run(cmd + pwd_cmd, stdout=subprocess.PIPE).stdout.decode("utf-8")
```

Comment:
We don't need to use subprocess here. We can use it as a python module. Please see https://github.com/DissectMalware/XLMMacroDeobfuscator#library

Reply:
Yeah, you are right